src/Raven.Server/Documents/Handlers/StreamCsvDocumentQueryResultWriter.cs

```diff
@@ -59,8 +60,15 @@ public void AddResult(Document res)
             WriteCsvHeaderIfNeeded(res);
             foreach (var property in _properties)
             {
-                var o = new BlittablePath(property).Evaluate(res.Data, false);
-                _csvWriter.WriteField(o?.ToString());
+                if (property.Length == 3 && property.Equals(Constants.Documents.Metadata.Id))
```

Comment:
Why do the `property.Length` check

Reply:
Most of the time its going to be false and comparing ints is cheaper than strings

Reply:
That is already something that is checked by the string anyway, no point in doing this.

Reply:
There is a point since it will get inlined and you save a method invocation

Reply:
There is a point since it will get inlined and you save a method invocation

Reply:
No, it isn't inlined in the sense you are talking about, and at any rate, this defeats the point of using a constant. See the blog post for the details

Reply:
The length property is inlined the string equal isn't. You're wrong. If you'd argue that the branch perdition will probably do a better job in detecting it's usually false it would be a stronger argument...

Reply:
The length property is inlined the string equal isn't. You're wrong. If you'd argue that the branch perdition will probably do a better job in detecting it's usually false it would be a stronger argument...

Reply:
The length property check is done by eventually going here: 
https://github.com/dotnet/coreclr/blob/238907f69662dda53dee0afd9848756a5bb790e9/src/classlibnative/bcltype/stringnative.cpp#L311

There are some patterns that are recognized by the JIT, as explained here:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/String.cs#L201

But you are going to end up doing a call anyway. At this point, it doesn't matter if you are going to run a few more instructions. The check happens anyway, here: 

https://github.com/dotnet/coreclr/blob/master/src/utilcode/sstring.cpp#L1597

There is also the issue of hard coding the length of `Constants.Documents.Metadata.Id`